src/Symfony/Component/DependencyInjection/Secrets/JweHandler.php

```diff
@@ -0,0 +1,298 @@
+<?php
+
+namespace Symfony\Component\DependencyInjection\Secrets;
+
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+use Symfony\Component\DependencyInjection\Exception\RuntimeException;
+
+final class JweHandler
+{
+    private $secretsLocation;
+
+    private $publicKey;
+
+    private $privateKeyLocation;
+
+    private $publicKeyLocation;
+
+    private $secrets;
+
+    /**
+     * @param string $fileLocation
+     *
+     * @throws InvalidArgumentException if the fileLocation does not exist in a writable directory
+     */
+    public static function initSecretsFile(string $fileLocation)
+    {
+        if (is_writeable(pathinfo($fileLocation, PATHINFO_DIRNAME))) {
+            file_put_contents($fileLocation, json_encode([], JSON_FORCE_OBJECT));
+        } else {
+            throw new InvalidArgumentException(sprintf(
+                'fileLocation for secrets at %s must exist in a writable directory',
+                $fileLocation
+            ));
+        }
+    }
```

Comment:
Instead of calling this method externally, could this class just call this internally whenever this class needs to write to taht file?

Reply:
yup, refactored to address